src/module/UserTermsCondition/user-terms-condition.services.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from "@nestjs/common";
+import { InjectModel } from "@nestjs/mongoose";
+import { UserTermsCondition } from "./user-terms-condition.schema";
+import { Model } from "mongoose";
+import { TermsCondition } from "../terms-condition/terms-condition.schema";
+
+@Injectable()
+export class UserTermsConditionService {
+    constructor(
+        @InjectModel('UserTermsCondition') private readonly userTermsConditionModal: Model<UserTermsCondition>,
+        @InjectModel('TermsCondition') private readonly termsConditionModel: Model<TermsCondition>
+    ) { }
+
+    async acceptTerms(userId: string, userAgent: string): Promise<UserTermsCondition> {
+        const activeTerms = await this.termsConditionModel.findOne({ active: true });
+        if (!activeTerms) throw new Error('No active terms found');
```

Comment:
Use `NotFoundException` it will retrun detailed error response

for your reference visit nestjs documents and collect the details about all type of nestjs exception